documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -0,0 +1,149 @@
+
+> [!TIP]
+> **Bottom line:** This option would be best if we want to retain control over the look and feel of the dashboard and manage the backend and frontend of the dashboard together, but can dedicate more engineering resources to building and hosting it and don't need to expose analytics data via API.
+
+- **Pros**
+  - Provides more fine-grained control over the look and feel of the dashboard application, on par with option 1.
+  - Allows a single engineer familiar with Python (or R if we use Shiny) to manage both the backend (e.g. analytics) and frontend (visualization and design of the dashboard).
+  - Faster to implement new charts and dashboards than an analytics API and separate frontend.
+  - Enables us to combine multiple dashboards into a single application.
+  - Allows open source contributors to host local versions of our dashboards.
+- **Cons**
+  - Building or modifying dashboards is still quite technical, not something a business analyst could do on their own.
+  - Harder to implement new charts or dashboards than SaaS or open source dashboard solution.
+  - Does not easily support adhoc reports or dashboards in the same tool.
+  - Does not easily support exposing analytics data to S2S users via API.
+
+### Open source dashboard solution
+
+This option involves selecting and hosting an open source dashboard solution, such as [Metabase][metabase] or [Redash][redash]. This solution would most likely be self-hosted and connect directly to our data warehouse, and enable business analysts to build adhoc reporting and dashboards with SQL and a drag-and-drop interface. Individual dashboards can then be configured for broader publication to external stakeholders.
+
+> [!TIP]
+> **Bottom line:** This option would be best if we want to adopt an open source solution that enables business analysts to build and host dashboards with minimal support from engineers, but are willing to compromise on the amount of control we have over the look and feel of those dashboards and don't need to expose analytics data via API.
+
+- **Pros**
+  - Enables a business analyst or engineer with basic SQL experience to build and manage dashboards.
+  - Faster and easier to implement new charts and dashboards than options 1 or 2.
+  - Supports adhoc reports and dashboards in the same tool.
+  - Allows open source contributors to host local versions of our dashboards (depending on the tool we choose).
+- **Cons**
+  - Harder to control the look and feel of the resulting dashboards.
+  - Harder to combine multiple dashboards into a single "application".
+  - Does not easily support exposing analytics data to S2S users via API.
```

Comment:
Hmm, I'm wondering if there are any options that do both. I know some SaaS BI tools offer "see the data behind this visual" over API or CSV download. Do either Metabase or Redash offer this?

Reply:
I did do a bit more research on this option, and linked to the API docs for Metabase and Redash in this commit: [fix: Clarifies pros + cons for dashboard solution options](https://github.com/HHS/simpler-grants-gov/pull/936/commits/8ab6dcf050dc4697eec405468ce870e9d9b8eb28)

The TL;DR is that at least some open source dashboard solutions may expose data via API, but the DX of accessing those endpoints is worse than our own custom analytics API and there may be some access/permission issues to think through.